### PR TITLE
Add implementation using Eventuate 

### PR DESCRIPTION
**Pull request by pkoryzna**

### CODE REVIEW

**Comment by lukasz-golebiewski** [comment]
Cool, I'll have a look at it later!

File: src/test/scala/esp/model/ApiTest.scala [hunk 1]
```diff
@@ -162,7 +162,8 @@ trait ApiTest extends FlatSpec with Matchers { api:Api =>
     val initialBankBalance = Bank.Account.balance
     val transferAmount: BigDecimal = 10
     api.transferMoney(accountNumber, api.listAccounts(id2).head, transferAmount)
-    Bank.Account.balance shouldBe initialBankBalance + Bank.transferFee(transferAmount)
+    api.getAccount(Bank.Account.number).map(_.balance) shouldBe Some(initialBankBalance + Bank.transferFee(transferAmount))
+    // how's that supposed to pass? Bank.Account is val!
```
Comment (lukasz-golebiewski) [inline src/test/scala/esp/model/ApiTest.scala]: Are you referring here to the deleted line?

File: src/main/scala/esp/eventuate/EventuateApi.scala [hunk 2]
```diff
@@ -1,35 +1,554 @@
...
+    case msg@GetTransactions(accountNumber) =>
+      val owner = userManager ? GetAccountOwnerId(accountNumber)
+      val snd = sender()
+      owner.onSuccess {
+        // forwarding the message with original sender because we're inside an async callback
+        case CommandSuccess(userId: UserId) => userManager.tell(GetUsersTransactions(accountNumber, userId), snd)
+        case CommandFailure(e: NoSuchElementException) => snd ! CommandSuccess(Seq.empty)
+      }
+
+    case t@TransferMoney(from, to, amount) =>
```
Comment (lukasz-golebiewski) [inline src/main/scala/esp/eventuate/EventuateApi.scala]: This case has high cyclomatic complexity. Could you refactor a bit?

File: src/test/scala/esp/model/ApiTest.scala (hunk 1 above)
Comment (pkoryzna) [inline src/test/scala/esp/model/ApiTest.scala]: Yes. I put it there as a mental note for later and left it in by mistake after modifying the test, apparently.

**Pull request closed without merging**